src/appleseed/renderer/modeling/input/sourceshadingpoint.h

```diff
@@ -0,0 +1,23 @@
+#ifndef APPLESEED_RENDERER_MODELING_INPUT_SOURCESHADINGPOINT_H
+#define APPLESEED_RENDERER_MODELING_INPUT_SOURCESHADINGPOINT_H
+
+// appleseed.foundation headers.
+#include "foundation/math/vector.h"
+
+// appleseed.main headers.
+#include "main/dllsymbol.h"
+
+namespace renderer
+{
+    class APPLESEED_DLLSYMBOL SourceShadingPoint
```

Comment:
I'd suggest `SourceInputs` instead, I think it would be clearer.

Don't forget to update the file names, the header guards...